Getting process diagram using /processes/{processDefinitionKey} fails if process has multiple versions

Using activiti-spring-boot-starter you can request a process diagramm using 
`GET /processes/{processDefinitionKey}.`
If multiple versions of the process are deployed, you receive an "Error 406 Not Acceptable".
Root cause: in ProcessEngineMvcEndpoint the singleResult() query fails with: 
`Caused by: org.activiti.engine.ActivitiException: Query return 2 results instead of max 1
	at org.activiti.engine.impl.AbstractQuery.executeSingleResult(AbstractQuery.java:187)
	at org.activiti.engine.impl.AbstractQuery.execute(AbstractQuery.java:166)
	at org.activiti.engine.impl.interceptor.CommandInvoker.execute(CommandInvoker.java:24)
	at org.activiti.engine.impl.interceptor.CommandContextInterceptor.execute(CommandContextInterceptor.java:57)
	at org.activiti.spring.SpringTransactionInterceptor$1.doInTransaction(SpringTransactionInterceptor.java:47)
	at org.springframework.transaction.support.TransactionTemplate.execute(TransactionTemplate.java:133)
	at org.activiti.spring.SpringTransactionInterceptor.execute(SpringTransactionInterceptor.java:45)
	at org.activiti.engine.impl.interceptor.LogInterceptor.execute(LogInterceptor.java:31)
	at org.activiti.engine.impl.cfg.CommandExecutorImpl.execute(CommandExecutorImpl.java:40)
	at org.activiti.engine.impl.cfg.CommandExecutorImpl.execute(CommandExecutorImpl.java:35)
	at org.activiti.engine.impl.AbstractQuery.singleResult(AbstractQuery.java:129)
	at org.activiti.spring.boot.actuate.endpoint.ProcessEngineMvcEndpoint.processDefinitionDiagram(ProcessEngineMvcEndpoint.java:45)
`

--- a/modules/activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-basic/src/main/java/org/activiti/spring/boot/actuate/endpoint/ProcessEngineMvcEndpoint.java
+++ b/modules/activiti-spring-boot/spring-boot-starters/activiti-spring-boot-starter-basic/src/main/java/org/activiti/spring/boot/actuate/endpoint/ProcessEngineMvcEndpoint.java
@@ -22,12 +22,16 @@ import org.springframework.boot.actuate.endpoint.mvc.EndpointMvcAdapter;
 import org.springframework.core.io.InputStreamResource;
 import org.springframework.core.io.Resource;
 import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 import org.springframework.web.bind.annotation.ResponseBody;
 
 import java.io.InputStream;
+import java.util.List;
+
+import static org.springframework.http.HttpStatus.NOT_FOUND;
 
 /**
  * Renders a valid running BPMN process definition as a BPMN diagram.
@@ -53,9 +57,17 @@ public class ProcessEngineMvcEndpoint extends EndpointMvcAdapter {
      */
     @RequestMapping(value = "/processes/{processDefinitionKey:.*}", method = RequestMethod.GET, produces = MediaType.IMAGE_JPEG_VALUE)
     @ResponseBody
-    public Resource processDefinitionDiagram(@PathVariable String processDefinitionKey) {
-        ProcessDefinition processDefinition = repositoryService.createProcessDefinitionQuery()
-                .processDefinitionKey(processDefinitionKey).singleResult();
+    public ResponseEntity processDefinitionDiagram(@PathVariable String processDefinitionKey) {
+        List processDefinitions = repositoryService.createProcessDefinitionQuery()
+                .processDefinitionKey(processDefinitionKey)
+                .orderByProcessDefinitionVersion()
+                .desc()
+                .list();
+        if (processDefinitions.isEmpty()) {
+            return ResponseEntity.status(NOT_FOUND).body(null);
+        }
+
+        ProcessDefinition processDefinition = (ProcessDefinition) processDefinitions.get(0);
         ProcessDiagramGenerator processDiagramGenerator = new DefaultProcessDiagramGenerator();
         BpmnModel bpmnModel = repositoryService.getBpmnModel(processDefinition.getId());
 
@@ -65,7 +77,6 @@ public class ProcessEngineMvcEndpoint extends EndpointMvcAdapter {
         }
 
         InputStream is = processDiagramGenerator.generateJpgDiagram(bpmnModel);
-        return new InputStreamResource(is);
+        return ResponseEntity.ok(new InputStreamResource(is));
     }
-
 }

